plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/CameraUpdateProvider.java

```diff
@@ -0,0 +1,64 @@
+package com.mapbox.mapboxsdk.plugins.locationlayer;
+
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdate;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+
+class CameraUpdateProvider implements RefreshPluginRunnable.RefreshPluginListener {
+
+  private final MapboxMap mapboxMap;
+  private final OnCameraMoveInvalidateListener onCameraMoveInvalidateListener;
+  private final RefreshPluginRunnable refreshPluginRunnable;
+  private float bearing = -1f;
+  private LatLng target;
+
+  CameraUpdateProvider(MapboxMap mapboxMap,
+                       OnCameraMoveInvalidateListener onCameraMoveInvalidateListener,
+                       RefreshPluginRunnable refreshPluginRunnable) {
+    this.mapboxMap = mapboxMap;
+    this.onCameraMoveInvalidateListener = onCameraMoveInvalidateListener;
+    this.refreshPluginRunnable = refreshPluginRunnable;
+    this.refreshPluginRunnable.setListener(this);
+  }
+
+  @Override
+  public void onShouldRefresh() {
+    CameraUpdate cameraUpdate = buildCameraUpdate();
+    mapboxMap.moveCamera(cameraUpdate);
+    onCameraMoveInvalidateListener.onInvalidateCameraMove();
+  }
+
+  void startUpdates() {
+    refreshPluginRunnable.run();
+  }
+
+  void cancelUpdates() {
+    refreshPluginRunnable.cancel();
+  }
+
+  void updateRefreshInterval(double refreshIntervalMillis) {
+    refreshPluginRunnable.updateRefreshInterval(refreshIntervalMillis);
+  }
+
+  void pushBearing(float bearing) {
+    this.bearing = bearing;
+  }
+
+  void pushLatLng(LatLng target) {
+    this.target = target;
+  }
+
+  private CameraUpdate buildCameraUpdate() {
+    CameraPosition cameraPosition = mapboxMap.getCameraPosition();
+    CameraPosition.Builder cameraPositionBuilder = new CameraPosition.Builder();
+    if (bearing > 0 && bearing != cameraPosition.bearing) {
+      cameraPositionBuilder.bearing(bearing);
+    }
+    if (target != null && !target.equals(cameraPosition.target)) {
```

Comment:
Similarly, as in https://github.com/mapbox/mapbox-plugins-android/pull/573/files#r200948002, we need to check if we are actually tracking the position and not only bearing, otherwise, we won't be able to move the camera.

Reply:
Maybe we can move those checks out of [LocationLayerCamera](https://github.com/mapbox/mapbox-plugins-android/blob/master/plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerCamera.java#L120-L133) and in here?